feature_details.md

```diff
@@ -0,0 +1,174 @@
+
+Cross-origin transitions have additional security/privacy constraints. In particular we must ensure the feature does not allow any cross-site information sharing which imposes the following restrictions :
+
+* The animations associated with the transition need to be defined completely by the previous Document, which is aware of the site the user is navigating to and has contextual information about the user journey.
+
+* The animations need to be executed such that there are no observable side-effects for the new and previous Document. For example, setting an obscure animation for a specific element on the new Document could be used to exchange information between origins. Enforcing this could influence the stage in the rendering pipeline where the UA executes these animations.
+
+An important assumption in the design of this feature is that it's reasonable to limit customization options for cross-origin transitions to make it easier for UAs to enforce the constraints above. But the API/implementation should strive to not unnecessarily limit capabilities for same-origin transitions due to cross-origin constraints.
+
+
+## Design
+**Disclaimer** : The details in this design focus on the SPA/MPA use-case with extensive customizability. It’s expected that the functionality will be delivered incrementally with the initial version of the API being quite limited in scope. The purpose of exploring the customizability options is to ensure the API can evolve to accommodate them.
+
+There are 2 kinds of semantic transitions which are a part of this feature :
+
+* Enter/Exit Transition : An element which exists only on the new Document animates as it enters the screen. The final state of the element in the new Document provides an end state for the animatable properties. Similarly an element which exists only on the previous Document animates as it exits the screen, with a start state for the animatable properties.
+
+* Paired Transition : An element which exists on both Documents is automatically animated from the start state on the previous Document to the end state on the new Document. The painted content for these elements may or may not be identical, which requires defining the behaviour for transitioning this content.
+
+### Control Flow
+This section covers the flow of events during a Document transition for the MPA use-case. The SPA scenario is identical except the events may map to different API points and are dispatched on the same Document. The code-snippets used are a rough API sketch with detailed explanation in subsequent sections.
+
+* The entry point for this API is document.documentTransition. The previous Document provides a url keyed list of elements, referred to as shared elements, which could be animated during a transition. There are a couple of important assumptions here :
```

Comment:
What is a url keyed list of elements? (probably a good addition for glossary section)